src/hera/workflows/v5/_mixins.py

```diff
@@ -232,155 +244,3 @@ def _build_volumes(self) -> Optional[List[Volume]]:
         if self.volumes is None:
             return None
         return [v._build_volume() for v in self.volumes]
-
-
-class _DAGTaskMixin(_BaseMixin):
```

Comment:
This was moved to a Task class since Task/Step have different fields and the callable syntactic sugar doesn't account for both. Deferring that for now and we can always implement syntactic sugar later.

It doesn't change much. Things go from

```py
task = echo(...)
```

to

```py
task = Task(echo)
# or
step = Step(echo)
```